src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -267,45 +267,37 @@ public void testCreateUserWithParams() throws Exception {
 
   @Test
   public void testLookupUserByPhone() throws Exception {
-    RandomUser randomUser1 = UserTestUtils.generateRandomUserInfo();
-    RandomUser randomUser2 = UserTestUtils.generateRandomUserInfo();
-    RandomUser randomUser3 = UserTestUtils.generateRandomUserInfo();
+    UserRecord user1 = null;
+    UserRecord user2 = null;
     try {
-      createUser(randomUser1.getUid(), /* phoneNumber= */ null, randomUser1.getEmail());
-      createUser(randomUser2.getUid(), randomUser2.getPhoneNumber(), randomUser2.getEmail());
-      importUser(randomUser3.getUid(), randomUser3.getPhoneNumber(), randomUser3.getEmail(),
-          "google.com", randomUser3.getUid() + "_google.com");
+      user1 = createRandomUser();
+      user2 = importRandomUser();
 
       UserRecord lookedUpRecord = auth.getUserByPhoneNumberAsync(
-          randomUser2.getPhoneNumber()).get();
-      assertEquals(randomUser2.getUid(), lookedUpRecord.getUid());
+          user1.getPhoneNumber()).get();
+      assertEquals(user1.getUid(), lookedUpRecord.getUid());
 
-      lookedUpRecord = auth.getUserByPhoneNumberAsync(randomUser3.getPhoneNumber()).get();
-      assertEquals(randomUser3.getUid(), lookedUpRecord.getUid());
+      lookedUpRecord = auth.getUserByPhoneNumberAsync(user2.getPhoneNumber()).get();
+      assertEquals(user2.getUid(), lookedUpRecord.getUid());
     } finally {
-      auth.deleteUserAsync(randomUser1.getUid()).get();
-      auth.deleteUserAsync(randomUser2.getUid()).get();
-      auth.deleteUserAsync(randomUser3.getUid()).get();
+      if (user1 != null) {
```

Comment:
I just remembered that we now have a `TemporaryUser` abstraction that handles automatic clean up of user accounts created during testing. Can we tie `createRandomUser()` and `importRandomUser()` methods to that abstraction, and clean up this logic?


Here's an example of how it's being used:
https://github.com/firebase/firebase-admin-java/blob/d8b1583002d60568106bf4a7ba2d5bcbbb6c0463/src/test/java/com/google/firebase/auth/FirebaseAuthIT.java#L446-L449

Reply:
Done.